### DISCUSSION TITLE
"Data Binding" example with typed attributes

### DISCUSSION THREAD

**Discussion by alexandre67fr** [livewire/livewire#4956] [7. Bugs (archived)] [answered]
Hello,

Let's take a look at the ["Data Binding" example from the documentation](https://laravel-livewire.com/docs/2.x/properties#data-binding)

```php
class HelloWorld extends Component
{
    public $message;
}
```

```html
<div>
    <input wire:model="message" type="text">
 
    <h1>{{ $message }}</h1>
</div>
```

So, we have a text input, and we bind a variable to it.

Now, let's assign type `string` to this variable, which kind of makes sense, since we're dealing with a text field :

```php
class HelloWorld extends Component
{
    public ?string $message;
}
```

This will throw an error `Typed property must not be accessed before initialization` when we start typing ([see other discussions about this error](https://github.com/livewire/livewire/issues?q=is%3Aissue+Typed+property+must+not+be+accessed+before+initialization))

Of course, we can do `public ?string $message = null;` or even `public string $message = '';`
 to fix this.

However, the problem is not the syntax of the component, but the `HandlesActions` trait : https://github.com/livewire/livewire/blob/master/src/ComponentConcerns/HandlesActions.php#L24

This trait checks if `$this->{$propertyName} instanceof Model`, but it doesn't check if `$this->{$propertyName}` is intialized, which is why the error occurs.

I'd propose to add `isset($this->{$propertyName})` check before `$this->{$propertyName} instanceof Model`, as shown here : https://github.com/alexandre67fr/livewire/commit/68a8e8d5bd354e97e6772a02c0baee0f48e4ccf1#diff-407939fdc8fd9ff8f7aca791faec0a01792b72a9f10ffc2bf215b2fa508fde46R24

I can create a MR if this makes sense.

Thanks

**Comment by joshhanley** [comment, marked as answer]
@alexandre67fr yep makes sense to me! PR's are welcome 🙂  please make sure to add tests

**Reply by alexandre67fr** [reply]
Sure, done :) https://github.com/livewire/livewire/pull/4957
